src/components/Chat/MessageList/index.js

```diff
@@ -6,21 +6,46 @@ import Message from './Message'
 const MessageList = props => {
   const { posts, currentUserId, getUserNamebyId, getIconColor } = props
 
+  let previousDate = 'not_defined'
+  let previousTime = 'not_defined'
+
   return (
     <div className="chat-message-list-container chat--message-list">
       <div className="chat--message-list--container">
         {posts.length > 0 &&
-          posts.map(post => (
-            <Message
-              key={post.id}
-              type={post.type}
-              sender={getUserNamebyId(post.user_id)}
-              text={post.message}
-              senderId={post.user_id}
-              currentUserId={currentUserId}
-              iconColor={getIconColor(post.user_id)}
-            />
-          ))}
+          posts.map(post => {
+            let showDate = false
```

Comment:
This might be cleaner as a separate function or two separate functions, for example getDateSent(post) and getShowDate(post).